dev/tools/localizations_validator.dart

```diff
@@ -24,7 +24,26 @@ void validateLocalizations(
   bool explainMissingKeys = false;
   for (final String locale in localeToResources.keys) {
     final Map<String, String> resources = localeToResources[locale];
-    final Set<String> keys = new Set<String>.from(resources.keys);
+
+    // Whether `key` corresponds to one of the plural variations of a key with
+    // the same prefix and suffix "Other".
+    //
+    // Many languages require only a subset of these variations, so we do not
+    // require them so long as the "Other" variation exists.
+    bool isPluralVariation(String key) {
```

Comment:
This looks fine, although it's N^2 for the plural resources. 

Maybe hoist the function out of the loop and make pluralRegexp a global? Not to remedy N^2, just to make the loop body a little smaller.

Reply:
The function closes over `resources`, so hoisting it up would require that I pass more context as parameters. I'm not worried about N^2 at all. This script runs for <0.5s. The advantage of having all the logic in one place outweighs greatly the performance benefits.